framework/src/timeintegrators/TimeIntegrator.C

```diff
@@ -42,6 +42,7 @@ TimeIntegrator::TimeIntegrator(const InputParameters & parameters)
     _n_nonlinear_iterations(0),
     _n_linear_iterations(0),
     _is_lumped(false),
+    _is_direct(isParamValid("use_direct") ? getParam<bool>("use_direct") : false),
```

Comment:
```suggestion
    _is_direct(false),
```

move `_is_direct = getParam<bool>("use_direct");` into CentralDifference